src/app/core/editing/session.js

```diff
@@ -322,44 +406,91 @@ proto._serializeCommit = function(itemsToCommit) {
-           if (value.properties[key] === undefined && childs_properties[key])
-             value.properties[key] = childs_properties[key]
-          }
-          const action = item.isNew() ? 'add' : item.getState();
-          // in case of add i have to remove non editable properties
-          layer[action].push(value);
-          break;
-      }
-    });
+    } else {
+      layer = commitObj;
+    }
+
+    items
+      .forEach((item) => {
+        state = item.getState();
+        const GeoJSONFormat = new ol.format.GeoJSON();
+        switch (state) {
+          case 'delete':
+            if (!item.isNew())
+              layer.delete.push(item.getId());
+            break;
+          default:
+            const value = GeoJSONFormat.writeFeatureObject(item);
+            const childs_properties = item.getProperties();
+            for (const key in value.properties) {
+             if (value.properties[key] && typeof value.properties[key] === 'object' && value.properties[key].constructor === Object)
+               value.properties[key] = value.properties[key].value;
+             if (value.properties[key] === undefined && childs_properties[key])
+               value.properties[key] = childs_properties[key]
+            }
+            const action = item.isNew() ? 'add' : item.getState();
+            // in case of add it have to remove not editable properties
+            layer[action].push(value);
+            break;
+        }
+      });
     // check in case of no edit remove relation key
     if (isRelation && !layer.add.length && !layer.update.length && !layer.delete.length) {
       delete commitObj.relations[key];
     }
   }
+
+  //get relation key (layerId) values
+  let relations = Object.keys(commitObj.relations);
+  // if not relations
+  if (relations.length > 0) {
+    //store relation layer id that belong to deep relations
+    const deleteRelationIds = [];
+    //Loop through relation layer id to commit
+    relations.forEach(rlId => {
+      //check if rlId (relation layer id) ia a child of current commit layer
+      if (
+        undefined === this._editor
+          .getLayer()
+          .getRelations()
+          .getArray()
+          .find(r => rlId === r.getChild())
+      ) {
+        //if not belong as child relation layer, check
+        const relationId = SessionsRegistry
+          .getSession(rlId)
+          .getEditor()
+          .getLayer()
+          .getRelations()
+          .getArray()
+          .find(r => relations.indexOf(r.getFather()) !== -1)
+          .getFather();
+
+        commitObj.relations[relationId].relations[rlId] = commitObj.relations[rlId];
+        //add to deleteRelationIds
+        deleteRelationIds.push(rlId);
+      }
+    })
+
+    //delete deep relations from current layer relation object
+    //because are not relative of this layer
+    deleteRelationIds.forEach(rId => delete commitObj.relations[rId]);
+  }
+
```

Comment:
@volterra79 I'm quite convinced it would be better to implement above changes after this other ones:

- https://github.com/g3w-suite/g3w-client-plugin-editing/pull/65
- https://github.com/g3w-suite/g3w-client/pull/493

Additionally, all editing-dependent plugins we have are already waiting for it to be established before starting (so it's no a real breaking change):

```
this.getDependencyPluginsObject(['editing']).then(..)
```

Please take a closer look you too, so we can start thinking decreasing all those (duplicated) editing related PRs..

Reply:
@Raruto i think is not possible to wait  because it need to pay attention and some test to check if plugins related to editing plugin works properly. Changes related to deep relation management need to be ready next week for g3w-suite v3.7.2